util/design/gen-otp-img.py

```diff
@@ -0,0 +1,183 @@
+
+                        Note that this seed must coincide with the seed used for enerating
+                        the OTP memory map (gen-otp-mmap.py).
+
+                        This value typically does not need to be specified as it is taken from
+                        the OTP memory map definition Hjson.
+                        ''')
+    parser.add_argument('-o',
+                        '--out',
+                        type=str,
+                        metavar='<path+filename>',
+                        default=MEMORY_HEXFILE,
+                        help='''
+                        Custom output path for generated hex file.
+                        Defaults to {}
+                        '''.format(MEMORY_HEXFILE))
+    parser.add_argument('--img-cfg',
+                        type=str,
+                        metavar='<path+filename>',
+                        default=IMAGE_DEFINITION_FILE,
+                        help='''
+                        Image configuration file in Hjson format.
+                        Defaults to {}
+                        '''.format(IMAGE_DEFINITION_FILE))
+    parser.add_argument('--add-cfg',
+                        type=str,
+                        metavar='<path+filename>',
+                        action='extend',
+                        nargs='+',
+                        help='''
+                        Additional image configuration file in Hjson format.
+
+                        This switch can be specified multiple times.
+                        Image configuration files are parsed in the same
+                        order as they are specified on the command line,
+                        and partition item values that are specified multiple
+                        times are overridden in that order.
+
+                        Note that seed values in additional configuration files
+                        are ignored.
+                        ''')
+
+    try:
+        args = parser.parse_args()
+
+        log.info('Loading LC state definition file {}'.format(
+            LC_STATE_DEFINITION_FILE))
+        with open(LC_STATE_DEFINITION_FILE, 'r') as infile:
+            lc_state_cfg = hjson.load(infile)
+        log.info('Loading OTP memory map definition file {}'.format(
+            MMAP_DEFINITION_FILE))
+        with open(MMAP_DEFINITION_FILE, 'r') as infile:
+            otp_mmap_cfg = hjson.load(infile)
+        log.info('Loading main image configuration file {}'.format(
+            args.img_cfg))
+        with open(args.img_cfg, 'r') as infile:
+            img_cfg = hjson.load(infile)
+
+        # If specified, override the seeds.
+        _override_seed(args, 'lc_seed', lc_state_cfg)
+        _override_seed(args, 'otp_seed', otp_mmap_cfg)
+        _override_seed(args, 'img_seed', img_cfg)
+
+        otp_mem_img = OtpMemImg(lc_state_cfg, otp_mmap_cfg, img_cfg)
+
+        print(args.add_cfg)
+
+        if args.add_cfg:
```

Comment:
I don't think you need this `if`: if the user didn't specify the argument, I think that `args.add_cfg` is `[]`.

Reply:
The argument comes back as `None` if not defined, hence this check.
However, I can make the argument default to `[]` in the argparse definition above, and then this can be omitted.

Reply:
Ah, I see. (I can never remember argparse defaults)